### PR TITLE
Show alphafold widget for a VEP-predicted variant

### PR DESCRIPTION
**Pull request by azangru**
### Background
This PR is the next step in the development of alphafold views on the current site. The expectation is that we will have three views, to parallel the current PDB 3d-protein views — specifically, a view for transcript, a view for variant, and a view for a VEP prediction.

These three varieties of alphafold views will be very similar in appearance; but will require different inputs, and will have to query different data sources, as is summarised in the table below:

|            | Data to start with                            | Client-side data fetching                                                                              |
|------------|-----------------------------------------------|--------------------------------------------------------------------------------------------------------|
| Transcript | <ul><li>species production name</li><li>protein stable id</li> | <ul><li>alphafold id</li><li>list of exons</li><li>list of protein features</li><li>list of variants</li></ul> |
| VEP        | <ul><li>species production name</li><li>protein stable id</li><li>alphafold id(?)</li><li>variant position</li><li>variant consequence</li></ul> | <ul><li>alphafold id(?)</li><li>list of exons</li><li>list of protein features</li></ul> |
| Variant    | <ul><li>species production name</li><li>variant id</li></ul> |  <ul><li>list of protein ids (which have alphafold analysis)</li><li>AF id for each protein id</li><li>list of exons</li><li>list of protein features</li><li>variant position</li><li>variant consequence</li></ul> |
  
Up until now, we have only had the alphafold view for transcript. This PR will add an alphafold view for VEP result; as well as refactor the existing code to allow for easier code reuse.

### General refactoring
- Moved the code responsible for the logic that will be shared across different variations of the alphafold viewer into their own files that act as "reactive controllers" in Lit. See the [documentation](https://lit.dev/docs/composition/controllers) if needed.

### Chores
- Updated Lit from v.2.0.0 to v.2.2.5
- Updated PDBe-molstar plugin from v.1.1.1 to v.3.0.0

### Additional visual changes
- Renamed the "Domains" column of VEP results table to "Protein matches"?
- Renamed the "Protein domains" dropdown in a VEP results table cell to "Protein matches"
- Added the "Alphafold model" button to show in VEP results table cells when appropriate
- Added the link to the Alphafold site for detected matches
- Changed the "Protein domains" label for the checkbox in the VEP input form to "Protein matches" and updated the text in the tooltip to mention Alphafold

### Notes
- We will need to ask Variation to update the description for the Domains column [here](https://github.com/Ensembl/ensembl-vep/blob/release/106/modules/Bio/EnsEMBL/VEP/Constants.pm#L179). We are using it [here](https://github.com/Ensembl/public-plugins/blob/release/106/tools/modules/EnsEMBL/Web/Component/Tools/VEP/Results.pm#L29)
- To be merged together with https://github.com/Ensembl/ensembl-webcode/pull/901 (which can be merged first)

### Jira ticket
https://www.ebi.ac.uk/panda/jira/browse/ENSWEB-6535

### Sandbox url
http://wp-np2-1e.ebi.ac.uk:8410/Homo_sapiens

Example VEP input: `8  26293684  rs1286787617  A  T`

### CODE REVIEW

File: tools/modules/EnsEMBL/Web/Component/Tools/VEP/Results.pm [hunk 1]
```diff
@@ -1265,6 +1251,18 @@ sub get_items_in_list {
...
+        # The alphafold ids we store in Ensembl databases (e.g. AF-P63151-F1.A)
+        # are fake ids, which consist of the real alphafold id followed by a dot and a chain name.
+        # Meanwhile, the alphafold site uses Uniprot ids as accession ids.
+        # A Uniprot id is the middle part of an alphafold id (separated by hyphens).
+        # Hopefully, things will improve in the future.
+        my $our_alphafold_id = $1;
+        my ( $actual_alphafold_id ) = $our_alphafold_id =~ /(.+)\./; # without the dot-chain
+        my ( $uniprot_id ) = $actual_alphafold_id =~ /-(.+)-/; # the middle part of an alphafold id
+
+        $item_url = "AFDB-ENSP_mappings:" . "&nbsp" . $hub->get_ExtURL_link($actual_alphafold_id, 'ALPHAFOLD', $uniprot_id);
```
Comment (ens-ma7) [inline tools/modules/EnsEMBL/Web/Component/Tools/VEP/Results.pm:1264]: i am confused why you need to pass `$actual_alphafold_id` in this function. Do we not need only the uniprot_id?

Example link:
https://alphafold.ebi.ac.uk/entry/P63151

**Comment by ens-ma7** [comment]
When clicking on the focus button to focus on the variant and then zooming out, there is something strange happening to the 3d structure as if some of the parts are not being drawn properly. Clicking the reset camera icon fix this and if i manually zoom in to the focus variant there is no issue with parts not being drawn. 

image

**Comment by azangru** [comment]
> When clicking on the focus button to focus on the variant and then zooming out, there is something strange happening to the 3d structure as if some of the parts are not being drawn properly. Clicking the reset camera icon fix this and if i manually zoom in to the focus variant there is no issue with parts not being drawn.

Yes, this is the expected behaviour.

File: widgets/modules/EnsEMBL/Web/Component/VEP/AFDB.pm [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  my $variant_label = $hub->param('var');
+  my $variant_position = $hub->param('pos');
+  my $variant_consequence = $hub->param('cons');
+
+  my $transcript  = $transcript_adaptor->fetch_by_stable_id($transcript_id);
+  my $translation = $transcript->translation();
+  my $translation_id = $translation->stable_id;
+
+  my $data_attributes = '';
+  $data_attributes .= qq{data-species="$species" }; # FIXME: won't be needed if not showing all variants
```
Comment (ens-ma7) [inline widgets/modules/EnsEMBL/Web/Component/VEP/AFDB.pm:50]: if the species that you are currently on that you need (from the URL), you dont need to pass this, there is a common function to get the species from the URL in the javascript. I cant remember the exact function but you can try this and see if it works:
`panel.elLk.species.val()`

File: tools/modules/EnsEMBL/Web/Component/Tools/VEP/Results.pm (hunk 1 above)
Comment (azangru) [reply tools/modules/EnsEMBL/Web/Component/Tools/VEP/Results.pm:1264]: `$actual_alphafold_id` is being used as text for the link (the part that goes in between the `<a>` tags). See the implementation of  `$hub->get_ExtURL_link` function.

**Review by ens-ma7 — APPROVED** [review]

**Pull request merged**